src/ports/friendship_synapse.rs

```diff
@@ -0,0 +1,79 @@
+use std::collections::HashMap;
```

Comment:
I know that I'd recommended adding this file to the `ports` folder, but do you think that this is the best place for it? Or should all the content of this file be in `src/components/synapse.rs`? Also, what should be the difference between ports and components? Maybe, `ports` was a bad abstraction that I introduced.

Reply:
Yeah, you're right. But instead of considering them as Synapse components, it'd make more sense to save their content in `src/entities/friendship_status.rs` and `src/entities/friendship_event.rs`

Reply:
Until now `entities` folder only contains abstractions of the models in the database, right? Do you think that this should be there too?